examples/stream_calc/tanslators/eventsub.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Translators that target the event publishing protocol."""
+
+from examples.stream_calc.ports.problem_receiver import ArithProblemReceiverPort
+from hexkit.custom_types import JsonObject
+from hexkit.protocols.eventsub import EventSubscriberProtocol
+
+
+class EventProblemReceiver(EventSubscriberProtocol):
+    """Inbound abstract translator translating between the EventSubscriberProtocol and the
+    ArithProblemReceiverPort."""
+
+    topics_of_interest = ["arithmetic_problems"]
+    types_of_interest = ["multiplication_problem", "devision_problem"]
+
+    def __init__(self, problem_receiver: ArithProblemReceiverPort):
+        """Configure the translator with a corresponding port implementation."""
+        self._problem_receiver = problem_receiver
+
+    def _check_payload(self, payload: JsonObject, excepted_fields: list[str]):
+        """Check whether all the expected fields are present in the payload."""
+        for field in excepted_fields:
+            if not field in payload:
```

Comment:
"not in" is more idiomatic (E713). Actually flake8 should complain about this, we should check that.

Reply:
Yes, I agree.